cfme/tests/cloud/test_instance_power_control.py

```diff
@@ -12,20 +12,9 @@
 
 
 def pytest_generate_tests(metafunc):
-    final_argv, final_ids = [], []
-
-    # Get all providers and pick those, that have power control test enabled
     argnames, argvalues, idlist = testgen.provider_by_type(
-        metafunc, ['ec2', 'openstack'], 'test_power_control')
-
-    for argn, argv, single_id in zip(argnames, argvalues, idlist):
-        test_pwr_ctl_i = argnames.index('test_power_control')
-        provider = argnames.index('provider')
-        if argv[test_pwr_ctl_i] is True:
-            final_argv.append(argv)
-            final_ids.append(argv[provider].key)
-
-    testgen.parametrize(metafunc, argnames, final_argv, ids=final_ids, scope="function")
+        metafunc, ['ec2', 'openstack'], required_fields=['test_power_control'])
```

Comment:
Im thinking maybe we could extend this to explicitly require the data to hold specific value if the key is not a collection...
E.g. `'test_power_control', True` or `'host', 'type', 'esxi'` or... `'version', '5'`.